src/s3filesystem.cpp

```diff
@@ -519,7 +519,7 @@ StatusCode S3FileSystem::downloadModelVersions(const std::string& path,
     const std::vector<model_version_t>& versions) {
     auto sc = createTempPath(local_path);
     if (sc != StatusCode::OK) {
-        SPDLOG_LOGGER_ERROR(s3_logger, "Failed to create a temporary path {}", sc);
+        SPDLOG_LOGGER_ERROR(s3_logger, "Failed to create a temporary path {}", Status(sc).string());
```

Comment:
We could use custom formaters:
https://github.com/gabime/spdlog#user-defined-types
Instead of polluting each call to spdlog with StatusCode/enums with casting